tests/test_db.py

```diff
@@ -578,6 +637,198 @@ def test_operations(self):
         self.assertEqual(op1_args['organization'], 'Example')
 
 
+class TestAddGroup(TestCase):
+    """"Unit tests for add_group"""
+
+    def setUp(self):
+        """Load initial dataset"""
+
+        self.user = get_user_model().objects.create(username='test')
+        self.ctx = SortingHatContext(self.user)
+
+        self.trxl = TransactionsLog.open('add_group', self.ctx)
+        self.orgname = "Example"
+        self.org = Organization.objects.create(name=self.orgname)
+
+    def test_add_group(self):
+        """Check if a new group is added"""
+
+        group_name = 'subgroup'
+        group = db.add_group(self.trxl, self.org, None, group_name)
+        self.assertIsInstance(group, Group)
+        self.assertEqual(group.name, group_name)
+        self.assertEqual(group.organization, self.org)
+
+    def test_add_subgroup(self):
+        """Check if a new subgroup is added for specified group"""
+
+        group_name = 'subgroup'
+        group = db.add_group(self.trxl, self.org, None, group_name)
+        subgroup = db.add_group(self.trxl, self.org, group, 'childgroup')
+        self.assertIsInstance(subgroup, Group)
+        self.assertEqual(subgroup.name, 'childgroup')
+        self.assertEqual(subgroup.organization, self.org)
```

Comment:
Check also that the parent is `subgroup`.

Reply:
Done 👍